app/src/main/java/com/example/cleanarchitecture/viewmodel/CharacterViewModel.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.cleanarchitecture.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.cleanarchitecture.utils.Data
+import com.example.cleanarchitecture.utils.Event
+import com.example.cleanarchitecture.utils.Status
+import com.example.domain.entity.CharacterDataWrapperEntity
+import com.example.domain.usecase.CharacterUseCase
+import com.example.domain.utils.Result
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+
+class CharacterViewModel(private val characterUseCase: CharacterUseCase) : ViewModel() {
+
+    private var mutableCharactersState: MutableLiveData<Event<Data<CharacterDataWrapperEntity>>> = MutableLiveData()
+    val liveDataState: LiveData<Event<Data<CharacterDataWrapperEntity>>>
+        get() = mutableCharactersState
+
+    fun onFetchCharactersClicked() {
+        viewModelScope.launch {
+            mutableCharactersState.value = Event(Data(responseType = Status.LOADING))
```

Comment:
the view is doing nothing for this state

Reply:
done